create-dmg

```diff
@@ -33,6 +33,26 @@ function pure_version() {
 	echo "$CDMG_VERSION"
 }
 
+function hdiutil-detach-retry() {
```

Comment:
```diff
- function hdiutil-detach-retry() {
+ function hdiutil_detach_retry() {
```

From my testing, functions with hyphens (or dashes) in their names work in both Bash 3.2.57 on macOS and 5.0.17 on Ubuntu, but I didn't find anywhere in the manual saying it is supported \* . To reduce confusion, I think it is better to use underscores.

\* : the manual says

```console
$ bash --version
GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)
...

$ man bash
...
DEFINITIONS
       The following definitions are used throughout the rest of this document.
       ...
       name   A word consisting only of alphanumeric characters and underscores, and  beginning  with  an  alphabetic
              character or an underscore.  Also referred to as an identifier.

...

   Shell Function Definitions
       A shell function is an object that is called like a simple command and executes a compound command with a  new  set
       of positional parameters.  Shell functions are declared as follows:

       name () compound-command [redirection]
       function name [()] compound-command [redirection]
              ...

```